regulation-worker/internal/delete/api/api.go

```diff
@@ -5,21 +5,80 @@ package api
 //called by delete/deleteSvc with (model.Job, model.Destination).
 //returns final status,error ({successful, failure}, err)
 import (
+	"bytes"
 	"context"
+	"encoding/json"
+	"fmt"
+	"net/http"
 
 	"github.com/rudderlabs/rudder-server/regulation-worker/internal/model"
 )
 
-type deleteManager interface {
-	Delete(ctx context.Context, job model.Job, destConfig map[string]interface{}, destName string) model.JobStatus
-}
 type API struct {
-	DeleteManager deleteManager
+	Client *http.Client
+	DestTransformURL string
 }
 
 //prepares payload based on (job,destDetail) & make an API call to transformer.
 //gets (status, failure_reason) which is converted to appropriate model.Error & returned to caller.
 func (api *API) Delete(ctx context.Context, job model.Job, destConfig map[string]interface{}, destName string) model.JobStatus {
 
-	return api.DeleteManager.Delete(ctx, job, destConfig, destName)
+	method := "DELETE"
+	endpoint := "/d-transformer/delete-users"
+	url := fmt.Sprint(api.DestTransformURL, endpoint)
+
+	bodySchema := mapJobToPayload(job, destName, destConfig)
+
+	reqBody, err := json.Marshal(bodySchema)
+	if err != nil {
+		return model.JobStatusFailed
+	}
+
+	req, err := http.NewRequestWithContext(ctx, method, url, bytes.NewReader(reqBody))
+	if err != nil {
+		return model.JobStatusFailed
+	}
+
+	// client := &http.Client{}
+	resp, err := api.Client.Do(req)
+	if err != nil {
+		return model.JobStatusFailed
```

Comment:
We are losing error information here, consider wrapping or logging

Reply:
yes, I am planning to add metrics & logging in complete GDPR code, once I am done with KV deleter

Reply:
added TODOs to add log error